### PR TITLE
feat(subscribe-form): add endpoint to handle form submission and point subscribe form to it

### PR DESCRIPTION
**Pull request by MichaelSolati**
Requires HubSpot private API key, I'm not sure where to include it but I stubbed a blank field in the .env file for now

### CODE REVIEW

File: components/Common/SubscribeForm/index.jsx [hunk 1]
```diff
@@ -101,41 +100,33 @@ const SubscribeForm = ({isCompactView}) => {
...
-        setFormSuccess(false);
-      } else if (data.result !== "success" || ! 'result' in data) {
-        setAfterFormSubmitMessage(('msg' in data ? data.msg : 'Something is wrong'));
+    try {
+      const response = await fetch(host, {
+        method: 'POST',
+        body: JSON.stringify(body)
+      });
+      const data = await response.json();
+      console.log(data)
```
Comment (yuval-hazaz) [inline components/Common/SubscribeForm/index.jsx:116]: please remove 
```suggestion
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
looks good, I will upload to staging so we can test it 
Please see minor comment

File: components/Common/SubscribeForm/index.jsx [hunk 2]
```diff
@@ -101,41 +100,33 @@ const SubscribeForm = ({isCompactView}) => {
     setAfterFormSubmitMessage('');
     setFormSuccess(null);
 
-    let host = 'https://dev.us5.list-manage.com/subscribe/post-json?';
-    let data = {
-      u: process.env.NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_U,
-      id: process.env.NEXT_PUBLIC_MAILCHIMP_ID,
-      c: process.env.NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_C,
+    const host = '/api/subscribe';
```
Comment (yuval-hazaz) [inline components/Common/SubscribeForm/index.jsx:103]: Should we use the env variable?

File: .env.development [hunk 3]
```diff
@@ -1,9 +1,6 @@
-NEXT_PUBLIC_MAILCHIMP_ID='39a8feead1'
-NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_U='cc391beefd07bdde0f03f8e95'
-NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_C='jQuery190022527047066005612_1646552810702'
-NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_='1646552810703'
 NEXT_PUBLIC_GITHUB_LINK='https://github.com/amplication/amplication'
 NEXT_PUBLIC_LINKEDIN_LINK='https://www.linkedin.com/company/amplication'
 NEXT_PUBLIC_TWITTER_LINK='https://twitter.com/amplication'
 NEXT_PUBLIC_SITE='https://amplication.com'
 NEXT_PUBLIC_MAIL='contact@amplication.com'
+NEXT_PRIVATE_HUBSPOT_API_KEY=''
```
Comment (yuval-hazaz) [inline .env.development:6]: what is the value?

File: .env.production [hunk 4]
```diff
@@ -1,9 +1,6 @@
...
-NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_C='jQuery190022527047066005612_1646552810702'
-NEXT_PUBLIC_MAILCHIMP_SUBSCRIBE_FORM_='1646552810703'
 NEXT_PUBLIC_GITHUB_LINK='https://github.com/amplication/amplication'
 NEXT_PUBLIC_LINKEDIN_LINK='https://www.linkedin.com/company/amplication'
 NEXT_PUBLIC_TWITTER_LINK='https://twitter.com/amplication'
 NEXT_PUBLIC_SITE='https://amplication.com'
-NEXT_PUBLIC_MAIL='contact@amplication.com'
\ No newline at end of file
+NEXT_PUBLIC_MAIL='contact@amplication.com'
+NEXT_PRIVATE_HUBSPOT_API_KEY=''
```
Comment (yuval-hazaz) [inline .env.production:6]: what is the value?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
we need the values for the hubspot API, please see my comments

File: components/Common/SubscribeForm/index.jsx (hunk 2 above)
Comment (MichaelSolati) [reply components/Common/SubscribeForm/index.jsx:103]: Submission to hubspot happens on server so this isn't necessary here.

File: .env.development (hunk 3 above)
Comment (MichaelSolati) [reply .env.development:6]: This is just stubbed for the API key we need. I didn't want to save it here but we will need to make sure it is saved/used by the site.

**Pull request merged**